### ISSUE TITLE
CruiseControl not starting Kafka 2.5.1 

### ISSUE THREAD

**Issue by punkrokk** [ninthnails/terraform-aws-camellia-image#4]
When using [ninthnails/terraform-aws-camellia-image](https://github.com/ninthnails/terraform-aws-camellia) terraform module, CruiseControl (since it's failing) with latest commits, I get the following error:
Kafka 2.5.1

```
[2020-10-16 20:58:57,889] priority=ERROR message="Uncaught exception on thread Thread[main,5,main]" category=com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain throwable="org.apache.kafka.common.config.ConfigException: Attempt to configure hard goals with unsupported goals (hard.goals:[com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal] and default.goals:[com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskUsageDistributionGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.TopicReplicaDistributionGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaDistributionGoal]).
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.sanityCheckGoalNames(KafkaCruiseControlConfig.java:171)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.<init>(KafkaCruiseControlConfig.java:427)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.<init>(KafkaCruiseControlConfig.java:422)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlUtils.readConfig(KafkaCruiseControlUtils.java:568)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain.main(KafkaCruiseControlMain.java:35)
"
[2020-10-16 20:58:57,889] priority=ERROR message="Uncaught exception on thread Thread[main,5,main]" category=com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain throwable="org.apache.kafka.common.config.ConfigException: Attempt to configure hard goals with unsupported goals (hard.goals:[com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal] and default.goals:[com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskUsageDistributionGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.TopicReplicaDistributionGoal, com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaDistributionGoal]).
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.sanityCheckGoalNames(KafkaCruiseControlConfig.java:171)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.<init>(KafkaCruiseControlConfig.java:427)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.<init>(KafkaCruiseControlConfig.java:422)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlUtils.readConfig(KafkaCruiseControlUtils.java:568)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain.main(KafkaCruiseControlMain.java:35)
"
```

The solution is to modify the cruisecontrol.properties file and ensure that `default.goals`, `goals` and `hard.goals` are all in sync. Wanted to see if @ninthnails is experiencing the same before I created a PR. My guess is it's now enforced by CruiseControl?

**Comment by ninthnails** [comment]
Thanks @punkrokk for pointing that out. I did came across issue trying to get 2.5.1 working and couldn't figure it out, hence I decided to post pone its support. If you want to create a PR, that would be great, I will have a look at it.

**Comment by punkrokk** [comment]
Roger that.

**Comment by ninthnails** [comment]
The issue has been fixed with release v1.5.2.
Merry Christmas.
